addon/dom-event-listeners.js

```diff
@@ -170,7 +170,7 @@ function getEventListenersDisposable(listeners) {
 
         element.removeEventListener(eventName, callback, options);
       }
-      listeners.length = 0;
+      eventListeners.set(obj, undefined);
```

Comment:
Why set to `undefined` (vs remove completely)?

Reply:
No particular reason, I got into this local maxima because I looked at the previous version which erroneously set `listeners = undefined` which just changed the reference but didn't update the thing which the `WeakMap` pointed to.

I'll take a swing at a real solution which actually takes advantage of the perf optimizations that exist here. (`WeakMap` target is an object with a `listeners` array.)